Installation error

Since `freetype-config` is not available on Archlinux the installation fails (actually the update to the latest version fails, so something changed there):
```
❯ LANG=C pip install aggdraw
Collecting aggdraw
  Using cached https://files.pythonhosted.org/packages/49/e8/5021fba0bf22434c1f15700f01fb67750a86bae0f320d39aee53a61ed968/aggdraw-1.3.5.tar.gz
Building wheels for collected packages: aggdraw
  Running setup.py bdist_wheel for aggdraw ... error
  Complete output from command /home/simon/.pyenv/versions/3.7.0/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/tmp/simon/pip-install-yez_bvvk/aggdraw/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/simon/pip-wheel-1qocje58 --python-tag cp37:
  Using ctypes to find freetype library...
  === freetype found: '/tmp/simon/pip-install-yez_bvvk'
  running bdist_wheel
  running build
  running build_ext
  building 'aggdraw' extension
  creating build
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/agg2
  creating build/temp.linux-x86_64-3.7/agg2/src
  creating build/temp.linux-x86_64-3.7/agg2/font_freetype
  ccache /usr/local/bin/cc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -DVERSION=1.3.5 -DHAVE_FREETYPE2 -Iagg2/include -Iagg2/font_freetype -I/tmp/simon/pip-install-yez_bvvk/include -I/tmp/simon/pip-install-yez_bvvk/include/freetype -I/tmp/simon/pip-install-yez_bvvk/include/freetype2 -I/home/simon/.pyenv/versions/3.7.0/include/python3.7m -c aggdraw.cxx -o build/temp.linux-x86_64-3.7/aggdraw.o
  In file included from aggdraw.cxx:84:
  agg2/font_freetype/agg_font_freetype.h:23:10: fatal error: ft2build.h: No such file or directory
   #include <ft2build.h>
            ^~~~~~~~~~~~
  compilation terminated.
  error: command 'ccache' failed with exit status 1
```

It could be related to #30, as `ctypes.util.find_library` does not return the full path of the so file (in my case it gives `'libfreetype.so.6'`):
https://github.com/pytroll/aggdraw/blob/fb26dedd70864e19f117dc24b1528328dc6d30b4/setup.py#L39-L40
So the flags are added but the paths are not valid.

--- a/setup.py
+++ b/setup.py
@@ -34,6 +34,16 @@ with the WCK renderer.
 """
 
 
+def _get_freetype_config():
+    print("Trying freetype-config to find freetype library...")
+    try:
+        # pointer to freetype build directory (tweak as necessary)
+        return subprocess.check_output(
+            ['freetype-config', '--prefix']).strip().replace(
+            b'"', b'').decode()
+    except (OSError, subprocess.CalledProcessError):
+        return None
+
 def _get_freetype_with_ctypes():
     print("Using ctypes to find freetype library...")
     from ctypes.util import find_library
@@ -41,16 +51,34 @@ def _get_freetype_with_ctypes():
     if ft_lib_path is None:
         return None
     ft_lib_path = os.path.dirname(ft_lib_path)
+    if not sys.platform.startswith('linux') and \
+            not os.path.isfile(ft_lib_path):
+        return None
+    elif not os.path.isfile(ft_lib_path):
+        # try prefix since find_library doesn't give a full path on linux
+        for bdir in (sys.prefix, '/usr', '/usr/local'):
+            lib_path = os.path.join(bdir, 'lib', ft_lib_path)
+            if os.path.isfile(lib_path):
+                return bdir
+
     lib_path = os.path.realpath(os.path.join(ft_lib_path, '..'))
     return lib_path
 
 
-try:
-    # pointer to freetype build directory (tweak as necessary)
-    FREETYPE_ROOT = subprocess.check_output(
-        ['freetype-config', '--prefix']).strip().replace(b'"', b'').decode()
-except (OSError, subprocess.CalledProcessError):
-    FREETYPE_ROOT = _get_freetype_with_ctypes()
+def _get_freetype_with_pkgconfig():
+    print("Trying 'pkgconfig' to find freetype library...")
+    try:
+        import pkgconfig
+        return pkgconfig.variables('freetype2')['prefix']
+    except (ImportError, KeyError, ValueError):
+        return None
+
+
+FREETYPE_ROOT = os.getenv('AGGDRAW_FREETYPE_ROOT')
+for func in (_get_freetype_config, _get_freetype_with_ctypes,
+             _get_freetype_with_pkgconfig):
+    if FREETYPE_ROOT is None:
+        FREETYPE_ROOT = func()
 
 if FREETYPE_ROOT is None:
     print("=== freetype not available")

